src/query/functions-v2/src/scalars/string.rs

```diff
@@ -106,6 +120,346 @@ pub fn register(registry: &mut FunctionRegistry) {
+                }
+            }
+            buff
+        },
+    );
+
+    registry.register_3_arg::<StringType, NumberType<u64>, StringType, StringType, _, _>(
+        "rpad",
+        FunctionProperty::default(),
+        |_, _, _| None,
+        |str: &[u8], l: u64, pad: &[u8]| {
+            let mut buff: Vec<u8> = vec![];
+            if l != 0 {
+                if l > str.len() as u64 {
+                    buff.extend_from_slice(str);
+                    while buff.len() < l as usize {
+                        if buff.len() + pad.len() <= l as usize {
+                            buff.extend_from_slice(pad);
+                        } else {
+                            buff.extend_from_slice(&pad[0..l as usize - buff.len()])
+                        }
+                    }
+                } else {
+                    buff.extend_from_slice(&str[0..l as usize]);
+                }
+            }
+            buff
+        },
+    );
+
+    registry.register_3_arg::<StringType, StringType, StringType, StringType, _, _>(
+        "replace",
+        FunctionProperty::default(),
+        |_, _, _| None,
+        |str, from, to| {
+            let mut buf: Vec<u8> = vec![];
+            if from.is_empty() || from == to {
+                buf.extend_from_slice(str);
+                return buf;
+            }
+            let mut skip = 0;
+            for (p, w) in str.windows(from.len()).enumerate() {
+                if w == from {
+                    buf.extend_from_slice(to);
+                    skip = from.len();
+                } else if p + w.len() == str.len() {
+                    buf.extend_from_slice(w);
+                } else if skip > 1 {
+                    skip -= 1;
+                } else {
+                    buf.extend_from_slice(&w[0..1]);
+                }
+            }
+            buf
+        },
+    );
+
+    registry.register_2_arg::<StringType, StringType, NumberType<i8>, _, _>(
+        "strcmp",
+        FunctionProperty::default(),
+        |_, _| None,
+        |s1, s2| {
+            let res = match s1.len().cmp(&s2.len()) {
+                Ordering::Equal => {
+                    let mut res = Ordering::Equal;
+                    for (s1i, s2i) in izip!(s1, s2) {
+                        match s1i.cmp(s2i) {
+                            Ordering::Equal => continue,
+                            ord => {
+                                res = ord;
+                                break;
+                            }
+                        }
+                    }
+                    res
+                }
+                ord => ord,
+            };
+            match res {
+                Ordering::Equal => 0,
+                Ordering::Greater => 1,
+                Ordering::Less => -1,
+            }
+        },
+    );
+
+    // registry.register_4_arg::<StringType, NumberType<i64>, NumberType<i64>, StringType, StringType, _, _>(
```

Comment:
Add 4_arg function. I'm not sure about the number of functions which has 4 args. If not too much I think maybe we just need to use register_function_factory impl the function?@andylokandy